lib/Controller/BanController.php

```diff
@@ -55,6 +58,17 @@ public function banActor(string $actorType, string $actorId, string $internalNot
 			$moderatorActorType = $moderator->getActorType();
 			$moderatorActorId = $moderator->getActorId();
 
+			$bannedParticipant = $this->participantService->getParticipantByActor($this->room, $actorType, $actorId);
+			
+			if ($bannedParticipant->getAttendee()->getParticipantType() === Participant::MODERATOR) {
+				throw new \InvalidArgumentException('Cannot ban a moderator');
+			}
+
+			if ($actorId === $moderatorActorId && $actorType === $moderatorActorType) {
+				throw new \InvalidArgumentException('Cannot ban yourself');
+			}
```

Comment:
Should be in the banService with `throw new \InvalidArgumentException('yourself')` and `'moderator'`

If you would keep it here, it should not throw but return a 400 data response instead, but that would just duplicate the work and allow to make mistakes if the services is invoked in another way at some point, so it's better to have the check in the service.